### PR TITLE
CDAP-2557: Move out non-handler logic from DatasetInstanceHandler

### PR DESCRIPTION
**Pull request by jwang47**
http://builds.cask.co/browse/CDAP-DUT1915-11

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+                                                          .addAll(props.getProperties())
+                                                          .build());
+    instanceManager.add(namespace, spec);
+
+    // Enable explore
+    enableExplore(newInstance, props);
+  }
+
+  /**
+   * Updates an existing Dataset specification properties  {@link DatasetInstanceConfiguration}
```
Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: Missing a period. Should be "Updates an existing Dataset specification properties."

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+   * is constructed based on request and the Dataset instance is updated.
+   *
+   * @throws DatasetNotFoundException if the dataset was not found
```
Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: missing parameters:
```
@param instance the dataset instance
@param  properties the dataset properties to be updated
```

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      // Type not found in the instance's namespace. Now try finding it in the system namespace
+      Id.DatasetType systemDatasetTypeId = Id.DatasetType.from(Constants.SYSTEM_NAMESPACE_ID, typeName);
+      typeMeta = implManager.getTypeInfo(systemDatasetTypeId);
+    }
+    return typeMeta;
+  }
+
+  /**
+   * Drops a dataset.
+   * @param spec specification of dataset to be dropped.
```
Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: Insert a blank line before the @param, and remove the trailing period on the parameter.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java [hunk 4]
```diff
@@ -82,30 +75,17 @@
...
-    this.implManager = implManager;
-    this.instanceManager = instanceManager;
-    this.exploreFacade = exploreFacade;
-    this.usageRegistry = usageRegistry;
-    this.allowDatasetUncheckedUpgrade = conf.getBoolean(Constants.Dataset.DATASET_UNCHECKED_UPGRADE);
+  public DatasetInstanceHandler(DatasetInstanceService instanceService) {
+    this.instanceService = instanceService;
   }
 
   @GET
```
Comment (anew) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java]: May I kindly ask that this class gets some Javadocs. I think it is important to document, for each handler method, what its path and query parameters mean.

**Comment by anew** [comment]
Overall I have quite a few concerns with this code. However, that is not subject of this PR, which appears to only move some logic from the HTTP handler into a new service. Which is cleaner and therefore better. What I would like to see is Javadocs that explain the non-obvious @PathParam and @QueryParam parameters of the handler methods. (For example the "owner" - what is that and what form does it have to have?)

Other than that LGTM.

**Pull request merged**